src/main/webapp/data-visualization.js

```diff
@@ -294,13 +301,28 @@ function setStyle(isCountyQuery) {
     const buttonsDiv = document.getElementById('buttons');
     buttonsDiv.style.display = 'none';
   }
-
   const chartsDiv = document.getElementById('am-charts');
   chartsDiv.style.display = 'block';
   const mapsDiv = document.getElementById('map');
   mapsDiv.style.display = 'none';
 }
 
+// Changes the color of the current visualizations on the page. 
+function changeColor(colorParam) {
+  if (typeof map !== 'undefined') {
+    const colorScale = chroma.scale(['white', colorParam.value]).domain([mapData.minValue,
+      mapData.maxValue]);
+    map.data.forEach(function(feature) {
+      map.data.setStyle((feature) => {
+        return {
+          fillColor: colorScale(mapData.map[feature.j.name]).toString(),
+        };
+      });
+    });
+  }
+  displayAmChartsMap(amChartsData, globalDescription, globalGeoData, colorParam.value);
```

Comment:
I think I see why you needed to use global variables, so that you could just call the displayAmChartsMap function again without re-doing every single aspect of the visualization. 
Here's an idea: what if we did basically this, but instead of having the global variables actually define what gets visualized (which seems risky), we make them be copies? So I'm essentially imagining that we have a global cache. The cache stores only one set of objects, the ones that were most recently created in displayVisualization. Then, we could have a displayCachedAmChartsMap function, which gets the variables out of the cache and sends them to  displayAmChartsMap. 
That way, inside displayVisualization, we're not working with global variables, we can just use local ones. But when everything is settled, we save it separately globally to be re-accessed if necessary.
This is just one approach so definitely interested to hear Kyra's thoughts as well.

Reply:
Changed to caching  all of the necessary data in localStorage. The google Maps api does not support directly caching maps.